llama-index-core/llama_index/core/utilities/token_counting.py

```diff
@@ -58,25 +59,33 @@ def estimate_tokens_in_messages(self, messages: List[ChatMessage]) -> int:
 
                 tokens += 3  # Additional tokens for function call
 
+            if "tool_calls" in additional_kwargs:
```

Comment:
isn't `tool_calls` specific to openai classes? a bit worried that this will be hard to generalize/debug

Reply:
It is, but this was already specific to openai (and most people use openai or use api's that copy openai)

A future PR would have to attach a token counter to the LLM I think, and make this the default